src/Umbraco.Web/PublishedContentQuery.cs

```diff
@@ -191,46 +191,44 @@ public IEnumerable<PublishedSearchResult> Search(string term, string culture = "
         /// <inheritdoc />
         public IEnumerable<PublishedSearchResult> Search(string term, int skip, int take, out long totalRecords, string culture = "*", string indexName = null)
         {
-            indexName = string.IsNullOrEmpty(indexName)
-                ? Constants.UmbracoIndexes.ExternalIndexName
-                : indexName;
+            if (string.IsNullOrEmpty(indexName))
+            {
+                indexName = Constants.UmbracoIndexes.ExternalIndexName;
+            }
 
             if (!_examineManager.TryGetIndex(indexName, out var index) || !(index is IUmbracoIndex umbIndex))
+            {
                 throw new InvalidOperationException($"No index found by name {indexName} or is not of type {typeof(IUmbracoIndex)}");
+            }
 
-            var searcher = umbIndex.GetSearcher();
-
-            // default to max 500 results
-            var count = skip == 0 && take == 0 ? 500 : skip + take;
+            var query = umbIndex.GetSearcher().CreateQuery(IndexTypes.Content);
 
-            ISearchResults results;
+            IQueryExecutor queryExecutor;
             if (culture == "*")
             {
-                //search everything
-
-                results = searcher.Search(term, count);
+                // Search everything
+                queryExecutor = query.ManagedQuery(term);
             }
-            else if (culture.IsNullOrWhiteSpace())
+            else if (string.IsNullOrWhiteSpace(culture))
```

Comment:
Because the extension method doesn't do anything else 😉 And the static method is always available (with auto-completion), without requiring the `Umbraco.Core` namespace or even Umbraco itself...

I would argue this makes it more readable, as it uses standard C# methods and you're sure it won't throw a `NullReferenceException` (with seems to be the biggest 'issue' of using this type of extension method: https://codereview.stackexchange.com/questions/6602/helper-class-for-null-and-empty-checks).

Reply:
Sure if you don't like it then don't use it ... but you changed what was already there :P anyways, it's fine.